overlord/assertstate/assertstate.go

```diff
@@ -356,15 +356,60 @@ func RefreshValidationSetAssertions(s *state.State, userID int) error {
 		return nil
 	}
 
-	if err := bulkRefreshValidationSetAsserts(s, vsets, userID, deviceCtx); err != nil {
-		return err
+	monitorModeSets := make(map[string]*ValidationSetTracking)
+	enforceModeSets := make(map[string]*ValidationSetTracking)
+	for vk, vset := range vsets {
+		if vset.Mode == Monitor {
+			monitorModeSets[vk] = vset
+		} else {
+			enforceModeSets[vk] = vset
+		}
+	}
+
+	monitorModeRefreshErr := bulkRefreshValidationSetAsserts(s, monitorModeSets, nil, userID, deviceCtx)
+
+	checkForConflicts := func(db *asserts.Database, bs asserts.Backstore) error {
+		vsets := snapasserts.NewValidationSets()
+		tmpDb := db.WithStackedBackstore(bs)
+		for _, vs := range enforceModeSets {
+			headers := map[string]string{
+				"series":     release.Series,
+				"account-id": vs.AccountID,
+				"name":       vs.Name,
+			}
+			var err error
+			var as asserts.Assertion
+			if vs.PinnedAt > 0 {
+				headers["sequence"] = fmt.Sprintf("%d", vs.PinnedAt)
+				as, err = tmpDb.Find(asserts.ValidationSetType, headers)
+			} else {
+				as, err = tmpDb.FindSequence(asserts.ValidationSetType, headers, -1, asserts.ValidationSetType.MaxSupportedFormat())
+			}
+			if err != nil {
+				return fmt.Errorf("internal error: cannot find validation set assertion: %v", err)
+			}
+
+			vsass, ok := as.(*asserts.ValidationSet)
+			if !ok {
+				return fmt.Errorf("internal error: unexpected assertion type %s for %s", vsass.Type().Name, ValidationSetKey(vs.AccountID, vs.Name))
+			}
+			if err := vsets.Add(vsass); err != nil {
+				return fmt.Errorf("internal error: cannot check validation sets conflicts: %v", err)
+			}
+		}
+		return vsets.Conflict()
+	}
+
+	enforceModeRefreshErr := bulkRefreshValidationSetAsserts(s, enforceModeSets, checkForConflicts, userID, deviceCtx)
```

Comment:
as we discussed if the error is exactly a conflict error we don't want to fail the overall operation, but want to log it and continue (the new validations were not committed though, so we can't update pointers for them below)